torch/testing/_internal/common_methods_invocations.py

```diff
@@ -7067,6 +7067,41 @@ def _gather(shape, index_dim, max_indices):
             )
         )
 
+    # Sample inputs to test edge cases for backward
+    # Check that gradients are propagated correctly for prod when zeros in self/src are reduced
+    if requires_grad and reduce == 'prod':
```

Comment:
this might be more cleanly modeled as a separate function for prod, that first yields from the common sample inputs for this class of ops and then extends the samples like this, but nbd